### ISSUE TITLE
no-undef rule in node env

### ISSUE THREAD

**Issue by rlidwka** [eslint/eslint#1333]
This fails with `'Object' is read only` message:

``` js
var Object = {}
```

This doesn't:

``` js
function() { var Object = {} }
```

This might make sense in browser. But in node.js environment all files are wrapped in IIFEs, so behavior should be the same:  `Object` is **not** read-only, it's a newly declared variable there.
